### ISSUE TITLE
Multi-channel input for annotate_with_whisper

### ISSUE THREAD

**Issue by WangHelin1997** [lhotse-speech/lhotse#865]
Hi, 
I found that only mono input is supported now for the `annotate_with_whisper` function. Will the multi-channel input work in the future? Or I only need to keep the first channel, what should I do? Thanks so much.

` if recording.num_channels > 1:
            logging.warning(
                f"Skipping recording '{recording.id}'. It has {recording.num_channels} channels, "
                f"but we currently only support mono input."
            )
            continue`

**Comment by pzelasko** [comment]
I think it should be straightforward to modify the code to iterate over every channel in a recording, create supervisions specifically for that channel, and bind it all together into a `MultiCut`. Would you be willing to make a PR with these improvements?

**Comment by pzelasko** [comment]
CC maybe @desh2608 would also be interested

**Comment by desh2608** [comment]
Yeah, that should be the most straightforward approach. If all channels are different instances of the same speech (i.e. same audio recorded with different mics, instead of 2 channels of a telephone conversations, for example), you can also just pick the first channel to transcribe and share the supervision for all channels. This would save a lot of compute.

**Comment by WangHelin1997** [comment]
Thanks for your help.
In addition, I found that annotate_with_whisper function takes very long time for mutiple audio files. Have you ever tested the speed of it?

**Comment by pzelasko** [comment]
I never optimized it for speed. There is a lot of discussion about it in the Whisper repo, maybe you can find something from there useful. I'd be happy to accept contributions with perf improvements :)

**Comment by desh2608** [comment]
At some point we may consider using one of the Python wrappers around [whisper.cpp](https://github.com/ggerganov/whisper.cpp/issues/9) instead of the original Whisper to speed things up. At the moment, it seems people have come up with several ways to wrap it in Python, such as with [cython](https://github.com/stlukey/whispercpp.py), [ctypes](https://github.com/ggerganov/whisper.cpp/issues/9#issuecomment-1342138572), and [pybind11](https://github.com/aarnphm/whispercpp). They seem promising, but most of them seem to have small issues such as incompatibility with Windows etc.

**Comment by entn-at** [comment]
> At some point we may consider using one of the Python wrappers around [whisper.cpp](https://github.com/ggerganov/whisper.cpp/issues/9) instead of the original Whisper to speed things up. At the moment, it seems people have come up with several ways to wrap it in Python, such as with [cython](https://github.com/stlukey/whispercpp.py), [ctypes](https://github.com/ggerganov/whisper.cpp/issues/9#issuecomment-1342138572), and [pybind11](https://github.com/aarnphm/whispercpp). They seem promising, but most of them seem to have small issues such as incompatibility with Windows etc.

I've actually started working on a second workflow that uses [faster-whisper](https://github.com/guillaumekln/faster-whisper/) powered by [CTranslate2](https://github.com/OpenNMT/CTranslate2)'s implementation (see https://github.com/lhotse-speech/lhotse/pull/1017). It's a lot faster and uses far less memory.

**Comment by desh2608** [comment]
Cool! Would you be interested in making a PR? I think it would be good to fall back to regular Whisper if people don't have the compute capability required for CTranslate2.
